Add React Native Hermes Profiling Alpha (#7228)
getsentry/sentry-docs: src/platforms/common/profiling/troubleshooting.mdx

Profiles for multi-threaded servers like `puma` might not capture frames correctly when async I/O is happening. This is a limitation of `stackprof`.

</PlatformSection>

<PlatformSection supported={["react-native"]}>

### Limitations

> Profile samples are collected at a fixed sampling frequency. We recommend checking if profiling's [Performance Overhead](/product/profiling/performance-overhead) is acceptable for your use case.
> 
> We are working on symbolication of the collected profiles. Out of the box, you will see function names.
> 
> As always, and especially since Profiling in React Native is an alpha feature, feedback is welcome on [Discord](https://discord.com/channels/621778831602221064/750735628932612096) or [GitHub](https://github.com/getsentry/sentry-react-native/issues/2668).

Reviewer: ```suggestion
Because profile samples are collected at a fixed sampling frequency, we recommend checking if profiling's [Performance Overhead](/product/profiling/performance-overhead) is acceptable for your use case. Note that 
we're working on symbolication of the collected profiles which means that you'll see function names out of the box.

As always, and especially since Profiling in React Native is an alpha feature, feedback is welcome on [Discord](https://discord.com/channels/621778831602221064/750735628932612096) or [GitHub](https://github.com/getsentry/sentry-react-native/issues/2668).
```

Author: ```suggestion
Because profile samples are collected at a fixed sampling frequency, we recommend checking if profiling's [Performance Overhead](/product/profiling/performance-overhead) is acceptable for your use case. Note that 
we're working on symbolication of the collected profiles which means that at the moment you'll see only function names but not resolved source file paths.

As always, and especially since Profiling in React Native is an alpha feature, feedback is welcome on [Discord](https://discord.com/channels/621778831602221064/750735628932612096) or [GitHub](https://github.com/getsentry/sentry-react-native/issues/2668).
```
